src/py/pyodide/console.py

```diff
@@ -479,6 +479,20 @@ async def runcode(self, source: str, code: CodeRunner) -> ConsoleFuture:
         return await super().runcode(source, code)
 
 
+def shorten(
+    text: str, limit: int = 1000, split: int | None = None, separator: str = "..."
+) -> str:
+    if limit < 2:
+        raise ValueError("limit must be greater than or equal to 4.")
+    if split is None:
+        split = limit // 2
+    if split > len(text) / 2:
```

Comment:
Maybe:

```suggestion
    split = min(split, len(text) // 2)
```

Reply:
I fixed it and updated the changelog as well. Thank you so much for kindly letting me know! Is there anything else I should do?

Reply:
I fixed it and updated the changelog as well. Thank you so much for kindly letting me know!